src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -397,250 +337,352 @@ export default function parseManifest(
+            }
+
+            if (representation.attributes.segmentProfiles != null) {
+              parsedRepresentation.segmentProfiles =
+                representation.attributes.segmentProfiles;
+            } else if (adaptation.attributes.segmentProfiles != null) {
+              parsedRepresentation.segmentProfiles =
+                adaptation.attributes.segmentProfiles;
+            }
+
+            if (representation.attributes.width != null) {
+              parsedRepresentation.width =
+                representation.attributes.width;
+            } else if (adaptation.attributes.width != null) {
+              parsedRepresentation.width =
+                adaptation.attributes.width;
+            }
+
+            return parsedRepresentation;
+          });
 
-          // 4-2-5. Add optional attributes
-          let codecs : string|undefined;
-          if (representation.attributes.codecs != null) {
-            codecs = representation.attributes.codecs;
-          } else if (adaptation.attributes.codecs != null) {
-            codecs = adaptation.attributes.codecs;
+        const adaptationMimeType = adaptation.attributes.mimeType;
+        const adaptationCodecs = adaptation.attributes.codecs;
+
+        const representationMimeTypes = representations
+          .map(r => r.mimeType)
+          .filter((mimeType : string|undefined) : mimeType is string => mimeType != null);
+
+        const representationCodecs = representations
+          .map(r => r.codecs)
+          .filter((codecs : string|undefined) : codecs is string => codecs != null);
+
+        const type: AdaptationType = inferAdaptationType(
+          adaptationMimeType || null,
+          representationMimeTypes,
+          adaptationCodecs || null,
+          representationCodecs,
+          adaptationChildren.role || null
+        ) as AdaptationType;
+
+        const mainAdaptation = (parsedAdaptations[type] || []).find((_adaptation) => {
+          if (
+            _adaptation.role === "main" &&
+            type === _adaptation.type
+          ) {
+            return true;
           }
+          return false;
+        });
 
-          if (codecs != null) {
-            codecs = codecs === "mp4a.40.02" ? "mp4a.40.2" : codecs;
-            parsedRepresentation.codecs = codecs;
-          }
+        let adaptationRole;
 
-          if (representation.attributes.audioSamplingRate != null) {
-            parsedRepresentation.audioSamplingRate =
-              representation.attributes.audioSamplingRate;
-          } else if (adaptation.attributes.audioSamplingRate != null) {
-            parsedRepresentation.audioSamplingRate =
-              adaptation.attributes.audioSamplingRate;
-          }
+        if (
+          adaptation.children.role &&
+          adaptation.children.role.value === "main" &&
+          adaptation.children.role.schemeIdUri === "urn:mpeg:dash:role:2011"
+        ) {
+          adaptationRole = "main";
```

Comment:
what is this compared to mainAdaptation and why the check is different? (schemeIdUri checked here)

Reply:
We check here if currently parsed adaptation is a 'main' one. On the other check, we rely on the already established adaptation role (see my response to the next comment).